lib/authenticate_methods.rb

```diff
@@ -50,35 +50,35 @@ def auto_login_external_user
     # Try to to find an existing user who has a matching external UUID
     # which indicates the account has been linked before
     # if found, we trust it and sign the user in automatically
-    if incoming_user.present?
-      sign_out!(security_log_data: {type: 'new external user'}) if signed_in?
-      sign_in!(incoming_user, security_log_data: {type: 'external uuid'})
-      return true
+    incoming_user = UserExternalUuid.find_by_uuid(external_user_uuid).try(:user)
+    return false if incoming_user.nil?
+
+    if signed_in? && incoming_user != current_user
+      # Sign out the current user; signing in the new user will effectively sign out
+      # the current user, but we choose to sign out explicitly so we can record
+      # information in the current user's security log
+
+      sign_out!(security_log_data: {type: 'different external user'})
     end
-    false
+
+    sign_in!(incoming_user, security_log_data: {type: 'external uuid'})
+    true
```

Comment:
This is the original logic that I think we still need.  We only want to `sign_out!` if the incoming user is different than the current user.